<a href="https://colab.research.google.com/github/Lizzy-TM/sistema-de-registro-2/blob/main/Curso_de_Estad%C3%ADstica_con_Python_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

***
# <font color=#56B4E9 size=10>CURSO DE ESTADÍSTICA</font>
***

<font color=#56B4E9 size = 5>*Preparando el ambiente*</font>

# <font color=#56B4E9>1 CONOCIENDO LOS DATOS</font>
***

## <font color=#56B4E9>1.1 Dataset del proyecto</font>
***

### Muestra de domicilios Colombia - 2018

Las investigaciones por muestras de domicilios realizadas anualmente, buscan encontrar característcias generales de la población, de educación, trabajo, rendimiento y otras, de acuerdo con las necesidades de información del país, tales como las características de migración, fertilidad, casamientos, salud, nutrición, entre otros temas. Estas muestras al pasar de los años consistuyen una herramienta importante para la formulación, validación y evaluación de políticas dirigidas al desarrollo socioeconómico y la mejora de las condiciones de vida en Colombia.


### Datos

Los datos fueron creados  de manera didáctica para este curso.

### Variables utilizadas

> ### Ingreso
> ***

Ingresos mensuales (en miles de pesos) del trabajo principal para personas de 10 años o más.

> ### Edad
> ***

Edad del entrevistado en la fecha de referencia en años.

> ### Altura
> ***

Altura del entrevistado em metros.

> ### Ciudad
> ***

Código de referência a 27 ciudades analizadas.


> ### Sexo
> ***

|Código|Descripción|
|---|---|
|0|Masculino|
|1|Femenino|

> ### Años de Estudio
> ***

|Código|Descripción|
|---|---|
|1|Sin estudios y menos de 1 año|
|2|1 año|
|3|2 años|
|4|3 años|
|5|4 años|
|6|5 años|
|7|6 años|
|8|7 años|
|9|8 años|
|10|9 años|
|11|10 años|
|12|11 años|
|13|12 años|
|14|13 años|
|15|14 años|
|16|15 años o más|
|17|No se sabe|
||No aplica|

> ### Color
> ***

|Código|Descripción|
|---|---|
|0|Indio|
|2|Blanco|
|4|Negro|
|6|Amarillo|
|8|Moreno|
|9|Sin declarar|

> ### Tratamiento a los datos
***

> Algunos de los tratamientos de datos más frecuentes son:
> 1. Eliminar las observaciones (líneas) con entradas de datos inválidos;
> 2. Eliminar observaciones donde hay datos perdidos (missing data);
> 3. Filtros própios de la investigación, por ejemplo: considerar solo las encuestas realizadas a la cabeza de familia (responsable por el domicilio).

### Importando pandas y leyendo el dataset del projecto

https://pandas.pydata.org/

In [22]:
import pandas as pd

In [23]:
datos = pd.read_csv("/content/datos.csv")

In [24]:
import pandas as pd

In [25]:
datos=pd.read_csv('/content/datos.csv')

In [26]:
datos.head()

Ciudad  Sexo  Edad  Color  Años de Estudio  Ingreso    Altura
0      11     0    23      8               12      800  1.603808
1      11     1    23      2               12     1150  1.739790
2      11     1    35      8               15      880  1.760444
3      11     0    46      2                6     3500  1.783158
4      11     1    47      8                9      150  1.690631

In [27]:
datos.sample(10)

Ciudad  Sexo  Edad  Color  Años de Estudio  Ingreso    Altura
920        11     0    35      8               12      900  1.808651
32813      31     0    59      2               16     9000  1.733597
15539      23     0    32      8               12      900  1.495389
65119      43     0    53      2                5     2700  1.746916
45711      33     0    28      2               12      800  1.743330
4338       13     0    44      8               12     2000  1.729812
35296      31     0    47      2               12      788  1.678761
18         11     0    57      2               12     2000  1.715138
5764       15     1    36      2               12      900  1.679054
69095      50     0    46      8                5     1500  1.746921

In [28]:
datos.sample(10)

Ciudad  Sexo  Edad  Color  Años de Estudio  Ingreso    Altura
64839      43     0    55      2               12     1800  1.721501
62163      43     1    35      2                9      280  1.641444
59828      42     0    68      2                4     1980  1.633649
62382      43     0    62      2                3      788  1.610266
9565       15     0    30      2               12     1300  1.759062
64803      43     1    27      2               16      500  1.747434
15231      23     1    21      2               12     1000  1.760558
33106      31     1    44      8                1      788  1.700573
13318      22     0    56      8                4      600  1.817774
49907      35     0    49      2                9     2210  1.564722

## <font color=#56B4E9>1.2 Tipos de datos</font>
***

### Clasificación de una variable
<img src='https://caelum-online-public.s3.amazonaws.com/ESP-1752-estadistica-con-python-parte1/img001.png' width='70%'>

### Variables cualitativas ordinales

► Variables que pueden ser ordenadas o que responen algun tipo de jerarquía

In [29]:
sorted(datos['Años de Estudio'].unique())

[np.int64(1),
 np.int64(2),
 np.int64(3),
 np.int64(4),
 np.int64(5),
 np.int64(6),
 np.int64(7),
 np.int64(8),
 np.int64(9),
 np.int64(10),
 np.int64(11),
 np.int64(12),
 np.int64(13),
 np.int64(14),
 np.int64(15),
 np.int64(16),
 np.int64(17)]

In [30]:
datos['Años de Estudio'].unique()

array([12, 15,  6,  9, 14, 16, 10,  1,  5,  7,  3,  2,  4,  8, 11, 17, 13])

In [31]:
sorted(datos['Años de Estudio'].unique())

[np.int64(1),
 np.int64(2),
 np.int64(3),
 np.int64(4),
 np.int64(5),
 np.int64(6),
 np.int64(7),
 np.int64(8),
 np.int64(9),
 np.int64(10),
 np.int64(11),
 np.int64(12),
 np.int64(13),
 np.int64(14),
 np.int64(15),
 np.int64(16),
 np.int64(17)]

### Variables cualitativas nominales

► Variables que no pueden ser ordenadas ni responden alguna jerarquía.

In [32]:
sorted(datos['Sexo'].unique())

[np.int64(0), np.int64(1)]

In [33]:
sorted(datos['Color'].unique())

[np.int64(0), np.int64(2), np.int64(4), np.int64(6), np.int64(8)]

In [34]:
sorted(datos['Color'].unique())

[np.int64(0), np.int64(2), np.int64(4), np.int64(6), np.int64(8)]

In [35]:
sorted(datos['Ciudad'].unique())

[np.int64(11),
 np.int64(12),
 np.int64(13),
 np.int64(14),
 np.int64(15),
 np.int64(16),
 np.int64(17),
 np.int64(21),
 np.int64(22),
 np.int64(23),
 np.int64(24),
 np.int64(25),
 np.int64(26),
 np.int64(27),
 np.int64(28),
 np.int64(29),
 np.int64(31),
 np.int64(32),
 np.int64(33),
 np.int64(35),
 np.int64(41),
 np.int64(42),
 np.int64(43),
 np.int64(50),
 np.int64(51),
 np.int64(52),
 np.int64(53)]

### Variables cuantitativas discretas

► Variables que representan un conteo donde los valores posibles forman un conjunto finito o numerable.


In [36]:
datos.Edad.max()

99

In [37]:
datos.Edad.unique()

array([23, 35, 46, 47, 34, 57, 60, 50, 26, 49, 52, 38, 45, 43, 53, 62, 69,
       41, 25, 33, 44, 65, 32, 30, 64, 31, 63, 40, 28, 37, 39, 36, 73, 56,
       27, 42, 51, 29, 24, 55, 67, 66, 61, 54, 48, 80, 79, 71, 59, 19, 22,
       77, 58, 17, 72, 70, 18, 21, 20, 68, 74, 76, 75, 84, 85, 78, 15, 82,
       95, 13, 83, 81, 87, 88, 16, 91, 92, 97, 89, 90, 86, 94, 14, 99])

In [38]:
datos.Edad.max()

99

In [39]:
datos['Edad'].max()

99

In [40]:
datos['Años de Estudio'].max()

17

In [41]:
datos.Años de Estudio.max()

SyntaxError: invalid syntax (ipython-input-41-2344663015.py, line 1)

In [ ]:
print(f"La edad mínima es {datos.Edad.min()} y la edad máxima es {datos.Edad.max()}")

In [ ]:
print(f"La edad mínima es {datos.Edad.min()} y la edad máxima es {datos.Edad.max()}")

#### <font color='red'>Observación</font>
***
> La variable edad puede ser clasificada de tres formas diferentes:
> 1. <b>CUANTITATIVA DISCRETA</b> - cuando representa años completos (números enteros);
> 2. <b>CUANTITATIVA CONTINUA</b> - cuando representa la edad exacta, siendo representada por fracciones de años;
> 3. <b>CUALITATIVA ORDINAL</b> - cuando representa  intervalos de edad.

### Variables cuantitativas continuas

► Variables que representan un conteo o medición que asume valores en escalas continuas (números reales).

In [ ]:
print(f"La Altura mínima es {datos['Altura'].min()} y la altura máxima es {datos['Altura'].max()}")

In [ ]:
print(f"La altura mínima es:{datos['Altura'].min()} y la altura máxima es:{datos['Altura'].max()}")

# <font color=#56B4E9>2 DISTRIBUCIÓN DE FRECUENCIAS</font>
***

El primer paso en un análisis es conocer el comportamiento de las variables involucradas en el estudio. Utilizando técnicas estadísticas como el análisis de las <b> DISTRIBUCIONES DE FRECUENCIAS </b> e <b> HISTOGRAMAS </b> podemos evaluar mejor cómo se distribuyen los fenómenos del estudio.



## <font color=#56B4E9>2.1 Distribución de frecuencias para variables cualitativas</font>
***

### Método 1

https://pandas.pydata.org/pandas-docs/version/0.22/generated/pandas.Series.value_counts.html

In [ ]:
datos.Sexo.value_counts()

In [ ]:
datos.Sexo.value_counts(normalize=True)

In [ ]:
datos.Sexo.value_counts(normalize=True)*100


In [ ]:
datos.Sexo.value_counts(normalize=True).round(4)*100

In [ ]:
frecuencia_sexo=datos.Sexo.value_counts()
frecuencia_sexo

In [ ]:
porcentaje_sexo=datos.Sexo.value_counts(normalize=True).round(4)*100
porcentaje_sexo

In [ ]:
pd.DataFrame({'Frecuencia':frecuencia_sexo,'Porcentaje %':porcentaje_sexo})

In [ ]:
dist_frec_cualitativa=pd.DataFrame({'Frecuencia':frecuencia_sexo,'Porcentaje %':porcentaje_sexo})

In [ ]:
dist_frec_cualitativa

In [ ]:
dist_frec_cualitativa.rename(index={0:'Masculino',1:'Femenino'},inplace=True)

In [ ]:
dist_frec_cualitativa

### Método 2

https://pandas.pydata.org/pandas-docs/version/0.22/generated/pandas.crosstab.html

In [ ]:
sexo = {0: 'Masculino',
        1: 'Femenino'}

color = {0: 'Indigena',
         2: 'Blanco',
         4: 'Negro',
         6: 'Amarillo',
         8: 'Pardo',
         9: 'Sin declarar'}

In [ ]:
frecuencia=pd.crosstab(datos.Sexo,datos.Color)
frecuencia

In [ ]:
frecuencia.rename(index=sexo,inplace=True)
frecuencia.rename(columns=color,inplace=True)
frecuencia

In [ ]:
porcentaje_sexo=datos.Sexo.value_counts(normalize=True).round(4)*100
porcentaje_color=datos.Color.value_counts(normalize=True).round(4)*100

In [ ]:
porcentaje=pd.crosstab(datos.Sexo,datos.Color,normalize=True)*100
porcentaje.rename(index=sexo,inplace=True)
porcentaje.rename(columns=color,inplace=True)
porcentaje

## <font color=#56B4E9>2.2 Distribución de frecuencia para variables cuantitativas (clases personalizadas)</font>
***

### Paso 1 - Especifique los límites de cada clase

Utilice la siguiente clasificación:

<b>A</b> ► Más de 20 SM

<b>B</b> ► De 10 a 20 SM

<b>C</b> ► De 4 a 10 SM

<b>D</b> ► De 2 a 4 SM

<b>E</b> ► Hasta 2 SM

donde <b>SM</b> es el valor del salario mínimo en ese momento. En nuestro caso <b>788 mil pesos colombianos</b> (2018):

<b>A</b> ► Más de 15.760

<b>B</b> ► De 7.880 a 15.760

<b>C</b> ► De 3.152 a 7.880

<b>D</b> ► De 1.576 a 3.152

<b>E</b> ► Hasta 1.576


In [ ]:
datos.Ingreso.min()

In [ ]:
datos.Ingreso.max()

In [ ]:
clases=[0,1576,3152,7880,15760,200000]

In [ ]:
label=['E','D','C','B','A']

### Paso 2 - Crear la tabla de frecuencias

https://pandas.pydata.org/pandas-docs/version/0.22/generated/pandas.cut.html

In [ ]:
frecuencia=pd.cut(x=datos.Ingreso,bins=clases,labels=label,include_lowest=True)

In [ ]:
frecuencia=pd.value_counts(pd.cut(x=datos.Ingreso,bins=clases,labels=label,include_lowest=True))
frecuencia

In [ ]:
porcentaje=pd.value_counts(pd.cut(x=datos.Ingreso,bins=clases,labels=label,include_lowest=True),normalize=True)*100
porcentaje

In [ ]:
porcentaje=pd.value_counts(pd.cut(x=datos.Ingreso,bins=clases,labels=label,include_lowest=True),normalize=True).round(4)*100
porcentaje

In [ ]:
dist_frec_cuantitativa_personalizada=pd.DataFrame({'Frecuencia':frecuencia,'Porcentaje %':porcentaje})
dist_frec_cuantitativa_personalizada

In [ ]:
dist_frec_cuantitativa_personalizada.sort_index(ascending=False)

## <font color=#56B4E9>2.3 Distribución de frecuencia para variables cuantitativas (clases de amplitud fija)</font>
***

### Importando bibliotecas

http://www.numpy.org/

In [78]:
import numpy as np


### Paso 1 - Difiniendo el número de clases

#### Regla de Sturges

# $$k = 1 + \frac {10}{3}\log_{10}n$$

In [79]:
datos.shape

(76840, 7)

In [80]:
n=datos.shape[0]
n

76840

In [81]:
k=1+((10/3)*np.log10(n))
k

np.float64(17.285291187298853)

### Paso 2 - Crear la tabla de frecuencias

In [ ]:
frecuencia_sturges=pd.value_counts(pd.cut(x=datos.Ingreso,bins=17,include_lowest=True))
frecuencia_sturges

In [ ]:
porcentaje_sturges=pd.value_counts(pd.cut(x=datos.Ingreso,bins=17,include_lowest=True),sort=False,normalize=True)*100
porcentaje_sturges

In [ ]:
dist_frec_cuantitativa_amplitud_fija=pd.DataFrame({'Frecuencia':frecuencia_sturges,'Porcentaje %':porcentaje_sturges})
dist_frec_cuantitativa_amplitud_fija

## <font color=#56B4E9>2.4 Histograma</font>
***

El <b>HISTOGRAMA</b> es la representación gráfica de una distribución de frecuencia. Es un gráfico formado por un conjunto de rectángulos colocados uno al lado del otro, donde el área de cada rectángulo es proporcional a la frecuencia de la clase que representa.

### Importando la biblioteca

https://seaborn.pydata.org/

In [ ]:
import seaborn as sns

In [ ]:
ax=sns.distplot(datos.Altura,kde=False)
ax.figure.set_size_inches(12,6)
ax.set_title('Distribución de frecuencias - Altura',fontsize=18)
ax.set_xlabel('Altura (metros)',fontsize=14)
ax

In [ ]:
ax=sns.histplot(datos.Altura,kde=True)
ax.figure.set_size_inches(12,6)
ax.set_title('Distribución de frecuencias - Altura -KDE',fontsize=18)
ax.set_xlabel('Altura (metros)',fontsize=14)
ax

# <font color=#56B4E9>3 MEDIDAS DE TENDENCIA CENTRAL</font>
***

## DataFrame de ejemplo

In [58]:
df = pd.DataFrame(data = {'María': [8, 10, 4, 8, 6, 10, 8],
                          'Pedro': [10, 2, 0.5, 1, 3, 9.5, 10],
                          'Pablo': [7.5, 8, 7, 8, 8, 8.5, 7]},
                  index = ['Matemática',
                           'Portugués',
                           'Inglés',
                           'Geografía',
                           'Historia',
                           'Física',
                           'Química'])
df.rename_axis('Asignaturas', axis = 'columns', inplace = True)
df

Asignaturas  María  Pedro  Pablo
Matemática       8   10.0    7.5
Portugués       10    2.0    8.0
Inglés           4    0.5    7.0
Geografía        8    1.0    8.0
Historia         6    3.0    8.0
Física          10    9.5    8.5
Química          8   10.0    7.0

## <font color=#56B4E9>3.1 Media aritmética</font>
***

Es representada por $\mu$ cuando se refiere a la población y por $\bar{X}$ cuando se refiere a la muestra

# $$\mu = \frac 1n\sum_{i=1}^{n}X_i$$

donde

$n$ = número de observaciones (registros)

$X_i$ = valor de la i-ésima observación (registro)

In [ ]:
(8+10+4+8+6+10+8)/7

In [59]:
df['María'].mean()

np.float64(7.714285714285714)

In [ ]:
datos.groupby(['Sexo'])['Ingreso'].mean()

## <font color=#56B4E9>3.2 Mediana</font>
***

Para obtener la mediana de un conjunto de datos, debemos proceder de la siguiente manera:
1. Ordenar el conjunto de datos;
2. Identificar el número de observaciones (registros) del conjunto de datos ($n$);
3. Identicar el elemento del medio:

> Cuando $n$ sea impar, la posición del elemento del medio se obtendrá de la siguiente manera:


# $$Elemento_{Md} = \frac{n+1}{2}$$

> Cuando $n$ sea par, la posición del elemento mediano se obtendrá de la siguiente manera:


# $$Elemento_{Md} = \frac{n}2$$

4. Obtener la mediana:

> Cuando $n$ sea impar:


# $$Md = X_{Elemento_{Md}}$$

> Cuando $n$ sea par:


# $$Md = \frac{X_{Elemento_{Md}} + X_{Elemento_{Md}+1}}2$$
***

### Ejemplo 1 - n impar

<img src='https://caelum-online-public.s3.amazonaws.com/ESP-1752-estadistica-con-python-parte1/img002.png' width='40%' style="float:left">

In [ ]:
notas_maria=df['María']
notas_maria

In [ ]:
notas_maria=notas_maria.sort_values()
notas_maria

In [ ]:
n=notas_maria.shape[0]
n

In [ ]:
notas_maria=notas_maria.reset_index()
notas_maria

In [ ]:
elemento_md=(n+1)/2
elemento_md


In [ ]:
notas_maria.loc[elemento_md-1]

In [ ]:
notas_maria['María'].median()

### Ejemplo 2 - n par

<img src='https://caelum-online-public.s3.amazonaws.com/ESP-1752-estadistica-con-python-parte1/img003.png' width='50%' style="float:left">

In [ ]:
notas_pedro=df['Pedro'].sample(6,random_state=200)
notas_pedro

In [ ]:
notas_pedro=notas_pedro.sort_values()
notas_pedro

In [ ]:
n=notas_pedro.shape[0]
n

In [ ]:
notas_pedro=notas_pedro.reset_index()
notas_pedro

In [ ]:
elemento_md=n/2
elemento_md

In [42]:
(notas_pedro.loc[elemento_md-1][1]+notas_pedro.loc[elemento_md][1])/2

NameError: name 'notas_pedro' is not defined

In [ ]:
notas_pedro['Pedro'].median()

### Obtener la mediana en nuestro conjunto de datos

## <font color=#56B4E9>3.3 Moda</font>
***

La moda se puede definir como el valor más frecuente de un conjunto de datos. La moda es ampliamente utilizada para datos cualitativos.

In [ ]:
df

In [ ]:
df.mode()

### Obteniendo la moda de nuestro dataset

In [ ]:
df.mode()


In [ ]:
ejemplo=pd.Series([1,2,2,3,4,4,5,6,7])
ejemplo

In [ ]:
ejemplo.mode()

In [ ]:
datos.Ingreso.mode()


In [ ]:
datos.Altura.mode()

## <font color=#56B4E9>3.4 Relación entre media, mediana e moda</font>
***

<img src='https://caelum-online-public.s3.amazonaws.com/ESP-1752-estadistica-con-python-parte1/img004.png' width='80%'>

### Evaluando la variable INGRESO

In [ ]:
ax=sns.distplot(datos.Ingreso)

In [ ]:
ax=sns.distplot(datos.query('Ingreso<20000').Ingreso)

In [ ]:
Moda=datos.Ingreso.mode()[0]
Moda

In [ ]:
Mediana=datos.Ingreso.median()
Mediana

In [ ]:
Media=datos.Ingreso.mean()
Media

In [ ]:
Moda<Mediana<Media

***

### Evaluando la variable ALTURA

In [43]:
Moda=datos.Altura.mode()[0]
Moda

np.float64(1.56812756)

In [44]:
Mediana=datos.Altura.median()
Mediana

1.6993247325

In [45]:
Media=datos.Altura.mean()
Media

np.float64(1.6995124540575741)

In [46]:
Moda<Mediana<Media

np.True_

***

### Evaluando la variable AÑOS DE ESTUDIO

In [47]:
Moda=datos["Años de Estudio"].mode()[0]
Moda

np.int64(12)

In [48]:
Mediana=datos["Años de Estudio"].median()
Mediana

11.0

In [49]:
Media=datos["Años de Estudio"].mean()
Media

np.float64(9.469664237376367)

In [50]:
Moda>Mediana>Media

np.True_

# <font color=#56B4E9>4 MEDIDAS DE LOCALIZACIÓN</font>
***

## <font color=#56B4E9>4.1 Cuartiles, deciles y percentiles</font>
***

Hay una serie de medidas de posición similares en su diseño a la mediana, aunque no son medidas de tendencia central. Como se sabe, la mediana divide la distribución en dos partes iguales en términos del número de elementos en cada parte. Los cuartiles, por otro lado, permiten que la distribución se divida en cuatro partes iguales en términos del número de elementos en cada uno; deciles en diez partes y centiles en cien partes iguales.

In [51]:
datos.Ingreso.quantile([0.25,0.5,0.75])

0.25     788.0
0.50    1200.0
0.75    2000.0
Name: Ingreso, dtype: float64

In [52]:
[i/10 for i in range(1,10)]

[0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9]

In [53]:
datos.Ingreso.quantile([i/10 for i in range(1,10)])

0.1     350.0
0.2     788.0
0.3     800.0
0.4    1000.0
0.5    1200.0
0.6    1500.0
0.7    1900.0
0.8    2500.0
0.9    4000.0
Name: Ingreso, dtype: float64

In [54]:
[i/100 for i in range(1,100)]

[0.01,
 0.02,
 0.03,
 0.04,
 0.05,
 0.06,
 0.07,
 0.08,
 0.09,
 0.1,
 0.11,
 0.12,
 0.13,
 0.14,
 0.15,
 0.16,
 0.17,
 0.18,
 0.19,
 0.2,
 0.21,
 0.22,
 0.23,
 0.24,
 0.25,
 0.26,
 0.27,
 0.28,
 0.29,
 0.3,
 0.31,
 0.32,
 0.33,
 0.34,
 0.35,
 0.36,
 0.37,
 0.38,
 0.39,
 0.4,
 0.41,
 0.42,
 0.43,
 0.44,
 0.45,
 0.46,
 0.47,
 0.48,
 0.49,
 0.5,
 0.51,
 0.52,
 0.53,
 0.54,
 0.55,
 0.56,
 0.57,
 0.58,
 0.59,
 0.6,
 0.61,
 0.62,
 0.63,
 0.64,
 0.65,
 0.66,
 0.67,
 0.68,
 0.69,
 0.7,
 0.71,
 0.72,
 0.73,
 0.74,
 0.75,
 0.76,
 0.77,
 0.78,
 0.79,
 0.8,
 0.81,
 0.82,
 0.83,
 0.84,
 0.85,
 0.86,
 0.87,
 0.88,
 0.89,
 0.9,
 0.91,
 0.92,
 0.93,
 0.94,
 0.95,
 0.96,
 0.97,
 0.98,
 0.99]

In [55]:
datos.Ingreso.quantile([i/100 for i in range(1,100)])

0.01        0.0
0.02        0.0
0.03        0.0
0.04       50.0
0.05      100.0
         ...   
0.95     6000.0
0.96     7000.0
0.97     8000.0
0.98    10000.0
0.99    15000.0
Name: Ingreso, Length: 99, dtype: float64

In [56]:
ax=sns.distplot(datos.Edad,hist_kws={'cumulative':True},
                kde_kws={'cumulative':True}, bins=10)
ax.figure.set_size_inches(14,6)
ax.set_title("Distribución de Frecuencia acumulada", fontsize=18)
ax.set_ylabel("Acumulado", fontsize=14)
ax.set_xlabel("Años", fontsize=14)
ax

NameError: name 'sns' is not defined

In [ ]:
datos.Edad.quantile([i/10 for i in range(1,10)])

## <font color=#56B4E9>4.2 Box-plot</font>
***

El *box plot* da una idea de la posición, dispersión, asimetría, colas y valores extremos. La posición central está dada por la mediana y la dispersión por $IIQ$ (distancia inter cuartil). Las posiciones relativas de $Q1$, $Mediana$ y $Q3$ dan una idea de la simetría de la distribución. Las longitudes de las colas están dadas por las líneas que van desde el rectángulo a los valores remotos y por los valores extremos.

<img src='https://caelum-online-public.s3.amazonaws.com/ESP-1752-estadistica-con-python-parte1/img005.png' width='65%'>

In [ ]:
ax=sns.boxplot(x='Altura',data=datos,orient='h')
ax.figure.set_size_inches(12,4)
ax.set_title("Altura", fontsize=18)
ax.set_xlabel("Metros", fontsize=14)
ax

In [ ]:
ax=sns.boxplot(x='Altura',y='Sexo',data=datos,orient='h')
ax.figure.set_size_inches(12,4)
ax.set_title("Altura", fontsize=18)
ax.set_xlabel("Metros", fontsize=14)
ax

In [ ]:
ax=sns.boxplot(x='Ingreso',data=datos,orient='h')
ax.figure.set_size_inches(12,4)
ax.set_title("Altura", fontsize=18)
ax.set_xlabel("Metros", fontsize=14)
ax

In [ ]:
ax=sns.boxplot(x='Ingreso',data=datos.query('Ingreso<10000'),orient='h')
ax.figure.set_size_inches(12,4)
ax.set_title("Altura", fontsize=18)
ax.set_xlabel("Metros", fontsize=14)
ax

In [ ]:
ax=sns.boxplot(x='Ingreso',y=("Sexo"),data=datos.query('Ingreso<10000'),orient='h')
ax.figure.set_size_inches(12,4)
ax.set_title("Ingreso", fontsize=18)
ax.set_xlabel("Miles de pesos", fontsize=14)
ax

In [ ]:
ax=sns.boxplot(x='Años de Estudio',data=datos,orient='h')
ax.figure.set_size_inches(12,4)
ax.set_title("Altura", fontsize=18)
ax.set_xlabel("Metros", fontsize=14)
ax

In [ ]:
ax=sns.boxplot(x='Años de Estudio',y='Sexo',data=datos,orient='h')
ax.figure.set_size_inches(12,4)
ax.set_title("Altura", fontsize=18)
ax.set_xlabel("Metros", fontsize=14)
ax

<img src='https://caelum-online-public.s3.amazonaws.com/ESP-1752-estadistica-con-python-parte1/img006.png' width='80%'>

# <font color=#56B4E9>5 MEDIDAS DE DISPERSIÓN</font>
***

Aunque las medidas de tendencia central y de localización proporcionan un resumen muy importante de los datos, pueden no ser suficientes para caracterizar diferentes conjuntos, especialmente cuando las observaciones de una distribución dada presentan datos muy dispersos.

## <font color=#56B4E9>5.1 Desviación media absoluta</font>
***


# $$DM = \frac 1n\sum_{i=1}^{n}|X_i-\bar{X}|$$


In [60]:
df

Asignaturas  María  Pedro  Pablo
Matemática       8   10.0    7.5
Portugués       10    2.0    8.0
Inglés           4    0.5    7.0
Geografía        8    1.0    8.0
Historia         6    3.0    8.0
Física          10    9.5    8.5
Química          8   10.0    7.0

In [63]:
notas_maria_dispersion=df[['María']]
notas_maria_dispersion

Asignaturas  María
Matemática       8
Portugués       10
Inglés           4
Geografía        8
Historia         6
Física          10
Química          8

In [65]:
nota_media_maria=notas_maria_dispersion.mean()[0]
nota_media_maria

/tmp/ipython-input-65-716702163.py:1: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  nota_media_maria=notas_maria_dispersion.mean()[0]


np.float64(7.714285714285714)

In [66]:
notas_maria_dispersion['Desv. Abs']=abs(notas_maria_dispersion['María']-nota_media_maria)
notas_maria_dispersion

/tmp/ipython-input-66-3096516480.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  notas_maria_dispersion['Desv. Abs']=abs(notas_maria_dispersion['María']-nota_media_maria)


Asignaturas  María  Desv. Abs
Matemática       8   0.285714
Portugués       10   2.285714
Inglés           4   3.714286
Geografía        8   0.285714
Historia         6   1.714286
Física          10   2.285714
Química          8   0.285714

In [67]:
notas_maria_dispersion['Desviacion']=(notas_maria_dispersion['María']-nota_media_maria)
notas_maria_dispersion

Asignaturas  María  Desv. Abs  Desviacion
Matemática       8   0.285714    0.285714
Portugués       10   2.285714    2.285714
Inglés           4   3.714286   -3.714286
Geografía        8   0.285714    0.285714
Historia         6   1.714286   -1.714286
Física          10   2.285714    2.285714
Química          8   0.285714    0.285714

In [71]:
notas_maria_dispersion['|Desviacion|']=notas_maria_dispersion['Desviacion'].abs()
notas_maria_dispersion

Asignaturas  María  Desv. Abs  Desviacion  |Desviacion|
Matemática       8   0.285714    0.285714      0.285714
Portugués       10   2.285714    2.285714      2.285714
Inglés           4   3.714286   -3.714286      3.714286
Geografía        8   0.285714    0.285714      0.285714
Historia         6   1.714286   -1.714286      1.714286
Física          10   2.285714    2.285714      2.285714
Química          8   0.285714    0.285714      0.285714

In [68]:
!pip install scipy

In [69]:
import scipy.stats as stats

In [70]:
mad_notas_maria=stats.median_abs_deviation(notas_maria_dispersion['María'])
mad_notas_maria

np.float64(2.0)

## <font color=#56B4E9>5.2 Varianza</font>
***

### Varianza

La varianza se construye a partir de las diferencias entre cada observación y la media de los datos, es decir, la desviación alrededor de la media. Al calcular la varianza, las desviaciones alrededor de la media son elevadas al cuadrado.

### Varianza de la población

# $$\sigma^2 = \frac 1n\sum_{i=1}^{n}(X_i-\mu)^2$$

### Varianza de la muestra

# $$S^2 = \frac 1{n-1}\sum_{i=1}^{n}(X_i-\bar{X})^2$$

In [72]:
notas_maria_dispersion

Asignaturas  María  Desv. Abs  Desviacion  |Desviacion|
Matemática       8   0.285714    0.285714      0.285714
Portugués       10   2.285714    2.285714      2.285714
Inglés           4   3.714286   -3.714286      3.714286
Geografía        8   0.285714    0.285714      0.285714
Historia         6   1.714286   -1.714286      1.714286
Física          10   2.285714    2.285714      2.285714
Química          8   0.285714    0.285714      0.285714

In [73]:
notas_maria_dispersion['(Desviacion)^2']=notas_maria_dispersion['|Desviacion|'].pow(2)
notas_maria_dispersion

Asignaturas  María  Desv. Abs  Desviacion  |Desviacion|  (Desviacion)^2
Matemática       8   0.285714    0.285714      0.285714        0.081633
Portugués       10   2.285714    2.285714      2.285714        5.224490
Inglés           4   3.714286   -3.714286      3.714286       13.795918
Geografía        8   0.285714    0.285714      0.285714        0.081633
Historia         6   1.714286   -1.714286      1.714286        2.938776
Física          10   2.285714    2.285714      2.285714        5.224490
Química          8   0.285714    0.285714      0.285714        0.081633

In [74]:
notas_maria_dispersion['(Desviacion)^2'].sum()/(len(notas_maria_dispersion)-1)

np.float64(4.57142857142857)

In [75]:
notas_maria_dispersion['María'].var()

4.57142857142857

## <font color=#56B4E9>5.3 Desviación estándar</font>
***

Una de las restricciones de la varianza es el hecho de que proporciona medidas cuadráticas de las unidades originales; la varianza de las medidas de longitud, por ejemplo, está en unidades de área. Por lo tanto, el hecho de que las unidades sean diferentes dificulta la comparación de la dispersión con las variables que la definen. Una forma de eliminar esta dificultad es considerar su raíz cuadrada.

### Desviación estándar de la población

# $$\sigma = \sqrt{\frac 1n\sum_{i=1}^{n}(X_i-\mu)^2} \Longrightarrow \sigma = \sqrt{\sigma^2}$$


### Desviación estándar de la muestra

# $$S = \sqrt{\frac 1{n-1}\sum_{i=1}^{n}(X_i-\bar{X})^2} \Longrightarrow S = \sqrt{S^2}$$

In [76]:
varianza=notas_maria_dispersion['María'].var()
varianza

4.57142857142857

In [82]:
np.sqrt(varianza)

np.float64(2.1380899352993947)

In [83]:
desviacion_standar=notas_maria_dispersion['María'].std()
desviacion_standar

2.1380899352993947

In [84]:
df.mean()

Asignaturas
María    7.714286
Pedro    5.142857
Pablo    7.714286
dtype: float64

In [85]:
df.median()

Asignaturas
María    8.0
Pedro    3.0
Pablo    8.0
dtype: float64

In [86]:
df.mode()

Asignaturas  María  Pedro  Pablo
0                8   10.0    8.0

In [88]:
df.std()

Asignaturas
María    2.138090
Pedro    4.460141
Pablo    0.566947
dtype: float64

Un grupo de 6 amigos registró el número de horas que durmieron la noche anterior. Los datos son los siguientes:

    7 horas
    6 horas
    8 horas
    7.5 horas
    6.5 horas
    8.5 horas

Calcula la desviación estándar de estas horas de sueño.

In [111]:
desviacion_amigos = pd.DataFrame(data = {'Horas': [7, 6, 8, 7.5, 6.5, 8.5]})
desviacion_amigos

Horas
0    7.0
1    6.0
2    8.0
3    7.5
4    6.5
5    8.5

In [113]:
desviacion_amigos.std()

Horas    0.935414
dtype: float64

In [112]:
desviacion_amigos.var()

Horas    0.875
dtype: float64

In [114]:
desviacion_amigos.mean()

Horas    7.25
dtype: float64